modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ViewObjectDefinitions/ViewObjectDefinitionsModals.tsx

```diff
@@ -236,6 +236,29 @@ export function ViewObjectDefinitionsModals({
 				/>
 			)}
 
+			{showModal.objectDefinitionOnRootModelDeletionNotAllowed &&
+				selectedObjectDefinition &&
+				Liferay.FeatureFlags['LPS-187142'] && (
+					<ModalObjectFieldDeletionNotAllowed
```

Comment:
Hey, this component `ModalObjectFieldDeletionNotAllowed` is named like it's only intended to control the deletion of object fields, but you are using here to control the deletion of object definitions. Maybe we should rename it to a more generic name?